File: projects/spring-boot-restv2/demo/src/main/java/com/sourabh/demo/events/CustomerModelListener.java
package com.sourabh.demo.events;

import com.sourabh.demo.model.Customer;
import com.sourabh.demo.service.SequenceGeneratorService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.mongodb.core.mapping.event.AbstractMongoEventListener;
import org.springframework.data.mongodb.core.mapping.event.BeforeConvertEvent;
import org.springframework.stereotype.Component;

@Component
public class CustomerModelListener extends AbstractMongoEventListener<Customer> {

    private SequenceGeneratorService sequenceGenerator;

    @Autowired
    public void customerModelListener(SequenceGeneratorService sequenceGenerator) {
        this.sequenceGenerator = sequenceGenerator;
    }

    @Override
    public void onBeforeConvert(BeforeConvertEvent<Customer> event) {
        if (event.getSource().getCid() < 1) {
            event.getSource().setCid(sequenceGenerator.generateSequence(Customer.SEQUENCE_NAME)+100);
        }
    }


}

File: projects/spring-boot-rest-kt/hotel-reservation/src/main/kotlin/com/sourabh/hotelreservation/events/CustomerModelListener.kt
package com.sourabh.hotelreservation.events

import com.sourabh.hotelreservation.model.Customer
import com.sourabh.hotelreservation.service.SequenceGeneratorService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.mongodb.core.mapping.event.AbstractMongoEventListener
import org.springframework.data.mongodb.core.mapping.event.BeforeConvertEvent
import org.springframework.stereotype.Component


@Component
class CustomerModelListener : AbstractMongoEventListener<Customer>() {
    private var sequenceGenerator: SequenceGeneratorService? = null
    @Autowired
    fun customerModelListener(sequenceGenerator: SequenceGeneratorService?) {
        this.sequenceGenerator = sequenceGenerator
    }

    override fun onBeforeConvert(event: BeforeConvertEvent<Customer>) {
        if (event.source.cid < 1) {
            event.source.cid=(sequenceGenerator!!.generateSequence(Customer.SEQUENCE_NAME) + 100)
        }
    }
}